qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -0,0 +1,363 @@
+        self.applied_gates = set()
+        while front_layer:
+            execute_gate_list = []
+
+            # Remove as many immediately applicable gates as possible
+            for node in front_layer:
+                if len(node.qargs) == 2:
+                    v0, v1 = node.qargs
+                    physical_qubits = (current_layout[v0], current_layout[v1])
+                    if physical_qubits in self.coupling_map.get_edges():
+                        execute_gate_list.append(node)
+                else:  # Single-qubit gates as well as barriers are free
+                    execute_gate_list.append(node)
+
+            if execute_gate_list:
+                for node in execute_gate_list:
+                    new_node = _transform_gate_for_layout(node, current_layout)
+                    mapped_dag.apply_operation_back(new_node.op,
+                                                    new_node.qargs,
+                                                    new_node.cargs,
+                                                    new_node.condition)
+                    front_layer.remove(node)
+                    self.applied_gates.add(node)
+                    for successor in dag.quantum_successors(node):
+                        if successor.type != 'op':
+                            continue
+                        if self._is_resolved(successor, dag):
+                            front_layer.append(successor)
+
+                    if node.qargs:
+                        self._reset_qubits_decay()
+
+                # Diagnostics
+                logger.debug('free! %s',
+                             [(n.name, n.qargs) for n in execute_gate_list])
+                logger.debug('front_layer: %s',
+                             [(n.name, n.qargs) for n in front_layer])
+
+                continue
+
+            # After all free gates are exhausted, heuristically find
+            # the best swap and insert it. When two or more swaps tie
+            # for best score, pick one randomly.
+            extended_set = self._obtain_extended_set(dag, front_layer)
+            swap_candidates = self._obtain_swaps(front_layer, current_layout)
+            swap_scores = dict.fromkeys(swap_candidates, 0)
+            for swap_qubits in swap_scores:
+                trial_layout = current_layout.copy()
+                trial_layout.swap(*swap_qubits)
+                score = self._score_heuristic(self.heuristic,
+                                              front_layer,
+                                              extended_set,
+                                              trial_layout,
+                                              swap_qubits)
+                swap_scores[swap_qubits] = score
+            min_score = min(swap_scores.values())
+            best_swaps = [k for k, v in swap_scores.items() if v == min_score]
+            best_swaps.sort(key=lambda x: (x[0].index, x[1].index))
+            best_swap = rng.choice(best_swaps)
```

Comment:
My understanding of `rng.choice` is that `best_swaps` will be shuffled.

```suggestion
            best_swap = rng.choice(best_swaps)
```

Reply:
that's fine, but i need to sort it so that a given seed behaves deterministically. otherwise it gives a different result each time (i tried it).